File: rackup/lib/core/protocol/actions.dart
// SYNC WITH: rackup-server/internal/protocol/actions.go

/// Action constants for WebSocket message routing.
abstract final class Actions {
  /// A player joined the lobby.
  static const String lobbyPlayerJoined = 'lobby.player_joined';

  /// A player left the lobby.
  static const String lobbyPlayerLeft = 'lobby.player_left';

  /// Full room state snapshot sent on WebSocket connect.
  static const String lobbyRoomState = 'lobby.room_state';

  /// A player submitted a punishment.
  static const String lobbyPunishmentSubmitted = 'lobby.punishment_submitted';

  /// A player's lobby status changed.
  static const String lobbyPlayerStatusChanged = 'lobby.player_status_changed';

  /// Host requests to start the game (client→server).
  static const String lobbyStartGame = 'lobby.start_game';

  /// Game has started — broadcast to all players (server→client).
  static const String lobbyGameStarted = 'lobby.game_started';

  /// Game initialized — server broadcasts full game state (server→client).
  static const String gameInitialized = 'game.initialized';

  /// A game turn was completed.
  static const String gameTurnComplete = 'game.turn_complete';

  /// Referee confirms a shot result (client→server, referee-only).
  static const String refereeConfirmShot = 'referee.confirm_shot';

  /// Referee undoes the last shot (client→server, referee-only).
  static const String refereeUndoShot = 'referee.undo_shot';

  /// Game has ended (server→client).
  static const String gameEnded = 'game.game_ended';

  /// An error occurred.
  static const String error = 'error';
}

File: rackup-server/internal/protocol/actions.go
package protocol

const (
	ActionLobbyPlayerJoined = "lobby.player_joined"
	ActionLobbyPlayerLeft   = "lobby.player_left"
	ActionLobbyRoomState    = "lobby.room_state"
	ActionGameTurnComplete  = "game.turn_complete"
	ActionLobbyPunishmentSubmitted  = "lobby.punishment_submitted"
	ActionLobbyPlayerStatusChanged = "lobby.player_status_changed"
	ActionLobbyStartGame           = "lobby.start_game"
	ActionLobbyGameStarted         = "lobby.game_started"
	ActionGameInitialized          = "game.initialized"
	ActionRefereeConfirmShot       = "referee.confirm_shot"
	ActionRefereeUndoShot          = "referee.undo_shot"
	ActionGameEnded                = "game.game_ended"
	ActionItemDeploy               = "item.deploy"    // client→server: player deploys an item
	ActionItemDeployed             = "item.deployed"  // server→client: broadcast item deployment
	ActionItemFizzled              = "item.fizzled"   // server→client: deployment failed (to deployer only)
	ActionError                    = "error"
)
